contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,59 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   void system_contract::addblnames( std::vector<name> blacklisted_name_patterns ) {
+      require_auth( get_self() );
+
+      check(!blacklisted_name_patterns.empty(), "Empty list of blacklisted name patterns provided");
+      check(blacklisted_name_patterns.size() <= 512, "Cannot provide more than 512 patterns in one action call");
+
+      account_name_blacklist_table bl_table(get_self(), get_self().value);
+      auto itr = bl_table.begin();
+      bool present = (itr != bl_table.end());
+
+      if (present) {
+         std::vector<name> current(itr->disallowed);
+
+         // number of blackcklist name patterns expected to be small, so quadratic check OK
+         for (auto n : blacklisted_name_patterns) {
+            if (std::find(std::cbegin(current), std::cend(current), n) == std::cend(current))
+               current.push_back(n);
+         }
+
+         bl_table.modify(itr, same_payer, [&](auto& blacklist) {
+            blacklist.disallowed = std::move(current);
+         });
```

Comment:
Could you avoid the copy by:

```c++
      if (present) {
         bl_table.modify(itr, same_payer, [&](auto& blacklist) {
            // number of blackcklist name patterns expected to be small, so quadratic check OK
            for (auto n : blacklisted_name_patterns) {
               if (std::find(std::cbegin(blacklist), std::cend(blacklist), n) == std::cend(blacklist))
                  blacklist.push_back(n);
            }
         });
```

Same for `rmblnames`.

Reply:
Yes, great suggestion. https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/c4b69681dc63c59ec4c3e06a19a54b0bcef6eb6e